### PR TITLE
Add example of Java - Spring - Spring JPA

### PR DESCRIPTION
**Pull request by gvdongen**
- Adds an example with Java and Spring
- Reads and writes to a postgres DB with Spring JPA

### CODE REVIEW

File: patterns-use-cases/integrations/java-spring/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,51 @@
+plugins {
```
Comment (slinkydeveloper) [inline patterns-use-cases/integrations/java-spring/build.gradle.kts:1]: Doesn't seem formatted?

File: patterns-use-cases/integrations/java-spring/src/main/java/dev/restate/examples/AppMain.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package dev.restate.examples;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+@SpringBootApplication
+public class AppMain {
```
Comment (slinkydeveloper) [inline patterns-use-cases/integrations/java-spring/src/main/java/dev/restate/examples/AppMain.java:7]: Is `AppMain` required? I recalled this is autoinjected somehow?

Comment (slinkydeveloper) [reply patterns-use-cases/integrations/java-spring/src/main/java/dev/restate/examples/AppMain.java:7]: in any case, let's just follow whatever start.spring.io generates.

**Review by slinkydeveloper — APPROVED** [review]
Can you run spotless on this project? Other than that, looks good to me

File: patterns-use-cases/integrations/java-spring/src/main/java/dev/restate/examples/AppMain.java (hunk 2 above)
Comment (gvdongen) [reply patterns-use-cases/integrations/java-spring/src/main/java/dev/restate/examples/AppMain.java:7]: This got generated by their template indeed. I am not aware of a way around this

**Pull request merged**